testsuite/integration/manualmode/src/test/java/org/jboss/as/test/manualmode/mgmt/elytron/HttpMgmtInterfaceElytronAuthenticationTestCase.java

```diff
@@ -0,0 +1,189 @@
+    }
+
+
+    @Test
+    @InSequence(0)
+    public void setupServer() throws Exception {
+        containerController.start(CONTAINER);
+        ManagementClient managementClient = getManagementClient();
+        host = Utils.getSecondaryTestAddress(managementClient, false);
+        prepareServerConfiguration(managementClient);
+    }
+
+    /**
+     * Test whether existing user with correct password has granted access through http-interface secured by Elytron
+     * http-authentication-factory.
+     */
+    @Test
+    @InSequence(1)
+    public void testCorrectUser() throws Exception {
+        Utils.makeCallWithBasicAuthn(createSimpleManagementOperationUrl(), USER, CORRECT_PASSWORD, SC_OK);
+    }
+
+    /**
+     * Test whether existing user with wrong password has denied access through http-interface secured by Elytron
+     * http-authentication-factory.
+     */
+    @Test
+    @InSequence(1)
+    public void testWrongPassword() throws Exception {
+        Utils.makeCallWithBasicAuthn(createSimpleManagementOperationUrl(), USER, "wrongPassword", SC_UNAUTHORIZED);
+    }
+
+    /**
+     * Test whether existing user with empty password has denied access through http-interface secured by Elytron
+     * http-authentication-factory.
+     */
+    @Test
+    @InSequence(1)
+    public void testEmptyPassword() throws Exception {
+        Utils.makeCallWithBasicAuthn(createSimpleManagementOperationUrl(), USER, "", SC_UNAUTHORIZED);
+    }
+
+    /**
+     * Test whether non-existing user has denied access through http-interface secured by Elytron http-authentication-factory.
+     */
+    @Test
+    @InSequence(1)
+    public void testWrongUser() throws Exception {
+        Utils.makeCallWithBasicAuthn(createSimpleManagementOperationUrl(), "wrongUser", CORRECT_PASSWORD, SC_UNAUTHORIZED);
+    }
+
+    /**
+     * Test whether user with empty username has denied access through http-interface secured by Elytron
+     * http-authentication-factory.
+     */
+    @Test
+    @InSequence(1)
+    public void testEmptyUser() throws Exception {
+        Utils.makeCallWithBasicAuthn(createSimpleManagementOperationUrl(), "", CORRECT_PASSWORD, SC_UNAUTHORIZED);
+    }
+
+    @Test
+    @InSequence(2)
+    public void resetServer() throws Exception {
+        try {
+            resetServerConfiguration(getManagementClient());
+        } finally {
+            containerController.stop(CONTAINER);
+        }
+    }
+
+    private ManagementClient getManagementClient() {
+        final ModelControllerClient client = TestSuiteEnvironment.getModelControllerClient();
+        return new ManagementClient(client, TestSuiteEnvironment.getServerAddress(),
+                TestSuiteEnvironment.getServerPort(), "http-remoting");
+    }
+
+    private URL createSimpleManagementOperationUrl() throws URISyntaxException, IOException {
+        return new URL("http://" + host + ":9990/management?operation=attribute&name=server-state");
```

Comment:
Does this work with IPv6? The JavaDoc of `Utils.getSecondaryTestAddress` says the square brackets are stripped.